apps/src/netsim/NetSimLobby.js

```diff
@@ -236,31 +236,45 @@ NetSimLobby.prototype.setShardID = function (shardID) {
  */
 NetSimLobby.prototype.onShardChange_ = function (shard, myNode) {
   // Unregister old handlers
-  if (this.eventKeys.nodeTable) {
-    this.shard_.nodeTable.tableChange.unregister(this.eventKeys.nodeTable);
-  }
-  if (this.eventKeys.wireTable) {
-    this.shard_.wireTable.tableChange.unregister(this.eventKeys.wireTable);
+  if (this.eventKeys.registeredShard) {
+    this.eventKeys.registeredShard.nodeTable.tableChange.unregister(
+        this.eventKeys.nodeTable);
+    this.eventKeys.registeredShard.wireTable.tableChange.unregister(
+        this.eventKeys.wireTable);
+    this.registeredShard = undefined;
```

Comment:
Here I'm switching to a newer, safer pattern I've been using, where we actually cache off the thing we need to unregister from.